geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateIndexCommand.java

```diff
@@ -78,118 +77,94 @@ public Result createIndex(@CliOption(key = CliStrings.CREATE_INDEX__NAME, mandat
     Result result;
     AtomicReference<XmlEntity> xmlEntity = new AtomicReference<>();
 
-    try {
-      IndexType idxType;
-
-      // Index type check
-      if ("range".equalsIgnoreCase(indexType)) {
-        idxType = IndexType.FUNCTIONAL;
-      } else if ("hash".equalsIgnoreCase(indexType)) {
-        idxType = IndexType.HASH;
-      } else if ("key".equalsIgnoreCase(indexType)) {
-        idxType = IndexType.PRIMARY_KEY;
-      } else {
-        return ResultBuilder
-            .createUserErrorResult(CliStrings.CREATE_INDEX__INVALID__INDEX__TYPE__MESSAGE);
-      }
-
-      if (indexName == null || indexName.isEmpty()) {
-        return ResultBuilder.createUserErrorResult(CliStrings.CREATE_INDEX__INVALID__INDEX__NAME);
-      }
-
-      if (indexedExpression == null || indexedExpression.isEmpty()) {
```

Comment:
Why is it safe to remove these validation check for `indexName` and `indexedExpression`?

Reply:
these are marked as "mandatory", so the parser should have already done the validation.  See the newly added CreateIndexCommandTest that covers these validations. 

Rule of thumb is: at the point (command exec body), we should have been beyond individual option validation already. This should only need to focus on the execution of the command. Validation is better done by the parser, the converter or the interceptor.

Reply:
Ah, I couldn't see the "mandatory" requirement in the collapsed diff. Looks good to me!